File: server/go/models/location.go
package models

import "time"

type Location struct {
	ID            uint      `gorm:"primaryKey" json:"id"`
	Active        bool      `json:"active"`
	Name          string    `gorm:"uniqueIndex;size:50;not null" json:"name"`
	Phone         string    `gorm:"size:15" json:"phone"`
	Address1      string    `gorm:"size:128" json:"address_1"`
	Address2      string    `gorm:"size:32" json:"address_2"`
	City          string    `gorm:"size:35" json:"city"`
	State         string    `gorm:"size:2" json:"state"`
	ZipCode       string    `gorm:"size:15" json:"zip_code"`
	DateCreated   time.Time `json:"date_created"`
	DateUpdated   time.Time `json:"date_updated"`
	CreatedUserID uint      `json:"created_user_id"`
	UpdatedUserID uint      `json:"updated_user_id"`
}

File: client/lib/models/location.dart
import 'dart:convert';

class Location {
  final int id;
  final bool? active;
  final String name;
  final String? phone;
  final String? address1;
  final String? address2;
  final String? city;
  final String? state;
  final String? zipCode;
  final DateTime dateCreated;
  final DateTime dateUpdated;
  final int? createdUserId;
  final int? updatedUserId;

  Location({
    required this.id,
    this.active,
    required this.name,
    this.phone,
    this.address1,
    this.address2,
    this.city,
    this.state,
    this.zipCode,
    required this.dateCreated,
    required this.dateUpdated,
    this.createdUserId,
    this.updatedUserId,
  });

  factory Location.fromJson(Map<String, dynamic> json) {
    return Location(
      id: json['id'],
      active: json['active'],
      name: json['name'],
      phone: json['phone'],
      address1: json['address_1'],
      address2: json['address_2'],
      city: json['city'],
      state: json['state'],
      zipCode: json['zip_code'],
      dateCreated: DateTime.parse(json['date_created']),
      dateUpdated: DateTime.parse(json['date_updated']),
      createdUserId: json['created_user_id'],
      updatedUserId: json['updated_user_id'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'active': active,
      'name': name,
      'phone': phone,
      'address_1': address1,
      'address_2': address2,
      'city': city,
      'state': state,
      'zip_code': zipCode,
      'date_created': dateCreated.toIso8601String(),
      'date_updated': dateUpdated.toIso8601String(),
      'created_user_id': createdUserId,
      'updated_user_id': updatedUserId,
    };
  }
}
